huaweicloud/services/evs/resource_huaweicloud_evs_snapshot_group.go

```diff
@@ -0,0 +1,517 @@
+			if err != nil {
+				return nil, "ERROR", err
+			}
+
+			if completed {
+				return "all snapshots completed", "COMPLETED", nil
+			}
+
+			return "snapshots still being created", "PENDING", nil
+		},
+		Timeout:      timeout,
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func checkVolumeSnapshotsStatus(client *golangsdk.ServiceClient, d *schema.ResourceData) (bool, error) {
+	queryParams := ""
+	if serverID, ok := d.GetOk("server_id"); ok {
+		queryParams += fmt.Sprintf("&server_id=%v", serverID)
+	}
+	if volumeIDs, ok := d.GetOk("volume_ids"); ok {
+		volumeIDList := volumeIDs.([]interface{})
+		if len(volumeIDList) > 0 {
+			// Convert volume IDs to JSON array format for API
+			volumeIDStrs := make([]string, len(volumeIDList))
+			for i, v := range volumeIDList {
+				volumeIDStrs[i] = v.(string)
+			}
+			// Use JSON array format for ids parameter
+			idsJSON, err := json.Marshal(volumeIDStrs)
+			if err != nil {
+				return false, fmt.Errorf("error marshaling volume IDs: %s", err)
+			}
+			queryParams += fmt.Sprintf("&ids=%s", string(idsJSON))
+		}
+	}
+
+	queryParams += "&status=creating"
+	requestPath := client.Endpoint + "v2/{project_id}/cloudvolumes/detail"
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	if queryParams != "" {
+		requestPath += "?" + queryParams[1:]
+	}
+
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	resp, err := client.Request("GET", requestPath, &requestOpt)
+	if err != nil {
+		return false, err
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return false, err
+	}
+
+	volumes := utils.PathSearch("volumes", respBody, []interface{}{}).([]interface{})
+
+	if len(volumes) > 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func refreshSnapshotGroupStatus(client *golangsdk.ServiceClient, snapshotGroupId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		respBody, err := GetSnapshotGroupDetail(client, snapshotGroupId)
+		if err != nil {
+			return nil, "ERROR", err
+		}
+		status := utils.PathSearch("snapshot_group.status", respBody, "").(string)
+		if status == "" {
+			return respBody, "ERROR", errors.New("status is not found in API response")
+		}
+		return respBody, status, nil
+	}
+}
+
+func handleMultiOperationsError(err error) (bool, error) {
+	if err == nil {
+		return false, nil
+	}
+	if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+		var apiError interface{}
+		if jsonErr := json.Unmarshal(errCode.Body, &apiError); jsonErr != nil {
+			return false, fmt.Errorf("unmarshal the response body failed: %s", jsonErr)
+		}
+		errorCode, errorCodeErr := jmespath.Search("error_code||errCode", apiError)
+		if errorCodeErr != nil {
+			return false, fmt.Errorf("error parse errorCode from response body: %s", errorCodeErr)
+		}
+		if errorCode == "EVS.2409" {
```

Comment:
what means "EVS.2409"? You need to explain it.
// If the response status code is 400 and the error code "EVS.2409" appears, it means that a snapshot is being created under the volume. We need to wait and try again.